src/compiler/crystal/macros/methods.cr

```diff
@@ -937,7 +937,7 @@ module Crystal
           when StringLiteral
             key = key.value
           else
-            return NilLiteral.new
+            raise "argument to [] must be a symbol or string, not #{key.class_desc}:\n\n#{key}"
```

Comment:
I'd suggest avoiding newlines in exception messages if possible.

Reply:
https://github.com/crystal-lang/crystal/pull/7153#discussion_r239688064